### PR TITLE
AzureMonitor: Traces updates

### PR DESCRIPTION
**Pull request by aangelisc**
- Fix distributed tracing by using fully qualified resource identifiers
- Add datalink for parent spans
- Fix trace logs explore view to redirect to Explore panel rather than Azure portal.

### CODE REVIEW

File: pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go [hunk 1]
```diff
@@ -680,7 +705,7 @@ func encodeQuery(rawQuery string) (string, error) {
 	return base64.StdEncoding.EncodeToString(b.Bytes()), nil
 }
 
-func buildTracesQuery(operationId string, traceTypes []string, filters []types.TracesFilters, resultFormat *string, resources []string) string {
+func buildTracesQuery(operationId string, parentSpanID *string, traceTypes []string, filters []types.TracesFilters, resultFormat *string, resources []string) string {
```
Comment (asimpson) [inline pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:708]: The placement of the arg is interesting here. Why not just append it to the end of the list of args?

File: pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go [hunk 2]
```diff
@@ -703,10 +728,8 @@ func buildTracesQuery(operationId string, traceTypes []string, filters []types.T
 	if len(resources) > 0 {
 		intermediate := make([]string, 0)
 		for _, resource := range resources {
-			resourceSplit := strings.SplitAfter(resource, "/")
-			resourceName := resourceSplit[len(resourceSplit)-1]
 			for _, table := range filteredTypes {
-				intermediate = append(intermediate, fmt.Sprintf("app('%s').%s", resourceName, table))
+				intermediate = append(intermediate, fmt.Sprintf("app('%s').%s", resource, table))
```
Comment (asimpson) [inline pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:732]: :heart: nice

**Review by asimpson — APPROVED** [review]
Nice refactor.

**Review by bossinc — APPROVED** [review]
:partying_face:

File: pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go (hunk 1 above)
Comment (aangelisc) [reply pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:708]: I feel like it just felt natural having it next to the `operationId` but I'm not picky 😄

Comment (asimpson) [reply pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go:708]: That was my guess! :raised_hands: Fine by me, just curious.

**Pull request merged**